assemblyline/odm/models/config.py

```diff
@@ -535,9 +535,11 @@ class Mount(odm.Model):
     privileged_only: bool = odm.Boolean(default=False,
                                         description="Should this mount only be available for privileged services?")
 
-    # ConfigMap-specific
-    config_map: str = odm.Optional(odm.Keyword(description="Name of ConfigMap (Kubernetes only)"))
-    key: str = odm.Optional(odm.Keyword(), description="Key of ConfigMap (Kubernetes only)")
+    # Kubernetes-specific
+    resource_type: str = odm.Enum(default='volume', values=['secret', 'configmap', 'volume'],
+                                  description="Type of mountable Kubernetes resource")
+    resource_name: str = odm.Keyword(description="Name of resource (Kubernetes only)")
+    resource_key: str = odm.Optional(odm.Keyword(), description="Key of ConfigMap/Secret (Kubernetes only)")
```

Comment:
This change breaks backward compatibility, we should bring back config_map and key but throw deprecation warnings in the scaler logs so we can eventually remove them.